### PR TITLE
[Hinty] Core typing: windows

### PR DESCRIPTION
**Pull request by gpotter2**
It's been a long time since I've done one of those.

- Core typing of Windows
- Change the mypy CI to test Linux AND Windows independently
- bunch of fixes everywhere, especially inconsistencies with the use of `NetworkInterface` or `str` in routes, that were sometimes messed up (but it worked because `NetworkInterface` can act as a string for backward compatibility)

### CODE REVIEW

File: .config/mypy/mypy.ini [hunk 1]
```diff
@@ -8,6 +8,9 @@ ignore_missing_imports = True
 
 # Layers specific config
 
+[mypy-scapy.arch.*]
+implicit_reexport = True
```
Comment (gpotter2) [inline .config/mypy/mypy.ini:12]: This is required because we have star imports in `scapy.arch` we can hardly get rid of (used for polyfill)

**Comment by gpotter2** [comment]
@guedou @p-l- This is ready for review. This is pretty interesting IMHO because it uses mypy's ability to specify the platform.

File: .config/mypy/mypy_check.py [hunk 2]
```diff
@@ -83,7 +96,23 @@
...
     ])
+    FILES = [
+        x for x in FILES if (
+            x not in [
+                # Disabled on Windows
+                "scapy/arch/linux.py",
+                "scapy/arch/solaris.py",
+                "scapy/contrib/cansocket_native.py",
+                "scapy/contrib/isotp/isotp_native_socket.py",
+            ]
```
Comment (p-l-) [inline .config/mypy/mypy_check.py:107]:
```suggestion
            x not in {
                # Disabled on Windows
                "scapy/arch/linux.py",
                "scapy/arch/solaris.py",
                "scapy/contrib/cansocket_native.py",
                "scapy/contrib/isotp/isotp_native_socket.py",
            }
```

**Review by p-l- — CHANGES_REQUESTED** [review]
Apart from a tiny change, LGTM. Thanks!

**Review by p-l- — APPROVED** [review]

**Pull request merged**